packages/stack/src/views/Stack/CardStack.tsx

```diff
@@ -378,6 +387,9 @@ export default class CardStack extends React.Component<Props, State> {
       onGestureStart,
       onGestureEnd,
       onGestureCancel,
+      detachInactiveScreens = shouldUseActivityState
+        ? Platform.OS !== 'ios'
+        : Platform.OS !== 'ios' && mode !== 'modal',
```

Comment:
```suggestion
      detachInactiveScreens = Platform.OS === 'ios'
        ? false // Disable `react-native-screens` on iOS by default since it's buggy
        : shouldUseActivityState || mode !== 'modal', // Enable on new versions of screens or for non modals on older versions
```